vault/provider.go

```diff
@@ -883,7 +883,18 @@ func MountCreateContextWrapper(f schema.CreateContextFunc, minVersion *version.V
 func importNamespace(d *schema.ResourceData) error {
 	if ns := os.Getenv(consts.EnvVarVaultNamespaceImport); ns != "" {
 		s := d.State()
-		if _, ok := s.Attributes[consts.FieldNamespace]; !ok {
+		var attemptNamespaceImport bool
+		if s == nil {
+			// state does not yet exist
+			// import is acceptable
+			attemptNamespaceImport = true
+		} else {
+			// only import if namespace
+			// is not already set in state
+			_, ok := s.Attributes[consts.FieldNamespace]
```

Comment:
I think we should hold the [lock](https://github.com/hashicorp/terraform-plugin-sdk/blob/35b0bd27f631b08fcf8e5a47ca4d411dde120359/terraform/state.go#L1317) before trying to access the [InstanceState](https://github.com/hashicorp/terraform-plugin-sdk/blob/35b0bd27f631b08fcf8e5a47ca4d411dde120359/terraform/state.go#L1284) fields.

Reply:
I am not sure if we would hit this scenario, but what do we do if the `Attributes` field is populated but the `ID` field is empty string? If we want to avoid importing the namespace in this scenario, maybe we could use the [Empty()](https://github.com/hashicorp/terraform-plugin-sdk/blob/35b0bd27f631b08fcf8e5a47ca4d411dde120359/terraform/state.go#L1402) method instead of the nil check above on L887?

Reply:
Ah good call! Forgot about that util. Updated in latest commit! 🙏🏼